tsdb/head_read.go

```diff
@@ -299,22 +313,36 @@ func (h *headChunkReader) Chunk(meta chunks.Meta) (chunkenc.Chunk, error) {
 	// This means that the chunk is outside the specified range.
 	if !c.OverlapsClosedInterval(h.mint, h.maxt) {
 		s.Unlock()
-		return nil, storage.ErrNotFound
+		return nil, 0, storage.ErrNotFound
+	}
+
+	chk, maxTime := c.chunk, c.maxTime
+	if headChunk && copyLastChunk {
+		// The caller may ask to copy the last chunk in order to take the
```

Comment:
I take this as a general comment. I understand that "head chunk" is confusing since it can be taken as a chunk that is in the Head block. Renaming all such occurrence and re-learning the new naming is out of scope at the moment, so calling it "head chunk" here as well for consistency.